src/schemas.jl

```diff
@@ -250,6 +253,9 @@ otherwise constitutes type piracy and should be avoided.
 @inline accepted_field_type(::SchemaVersion, T::Type) = T
 accepted_field_type(::SchemaVersion, ::Type{UUID}) = Union{UUID,UInt128}
 accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
+accepted_field_type(::SchemaVersion, ::Type{Vector{String}}) = AbstractVector{<:AbstractString}
```

Comment:
if this were true we'd want to define:

```
accepted_field_type(::SchemaVersion, ::Type{String}) = AbstractString
```

and e.g. make sure the `Symbol` definition above maps to `AbstractString` not `String`, etc.